src/poetry/puzzle/provider.py

```diff
@@ -458,7 +460,7 @@ def get_package_from_url(cls, url: str) -> Package:
         file_name = os.path.basename(urllib.parse.urlparse(url).path)
         with tempfile.TemporaryDirectory() as temp_dir:
             dest = Path(temp_dir) / file_name
-            download_file(url, dest)
+            download_file(url, dest, session=self._url_authenticator, chunk_size=1024 * 1024)
```

Comment:
The increased `chunk_size` is needed to ensure a fast read from the file cache when there's a cache hit. The default `chunk_size = 1024` would make the read pretty slow